src/Collectors/JobCollector.php

```diff
@@ -68,12 +79,23 @@ protected function setTransactionResult(string $transaction_name, int $result):
         ]);
     }
 
-    protected function stopTransaction(string $transaction_name): void
+    protected function stopTransaction(string $transaction_name, int $result): void
+    {
+        // Stop the transaction and measure the time
+        $this->agent->stopTransaction($transaction_name);
+        $this->agent->collectEvents($transaction_name);
+
+        $this->setTransactionResult($transaction_name, $result);
+    }
+
+    protected function send(Job $job): void
     {
         try {
-            // Stop the transaction and measure the time
-            $this->agent->stopTransaction($transaction_name);
-            $this->agent->collectEvents($transaction_name);
+            if (! ($job instanceof SyncJob)) {
+                // When using a queued driver, send/flush transaction to make room for the next job in the queue
+                // Otherwise just send when the agent destructs
+                $this->agent->send();
```

Comment:
I really wanted to do this in the `Looping` event. Unfortunately that runs before picking up a job, so no transaction has been started. And we can't check to see if there is a transaction because the transaction store is private, and the Looping event doesn't have a job in it so we can't check by name.